### PR TITLE
Adding search by title to NVI candidates.

### PR DESCRIPTION
**Pull request by EyobSikt**
Adding search by title to NVI candidates. Results give partial matches.

### CODE REVIEW

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/OpenSearchClientTest.java [hunk 1]
```diff
@@ -425,6 +462,13 @@ private static CandidateSearchParameters.Builder defaultSearchParameters() {
                    .withCustomer(CUSTOMER).withUsername(USERNAME).withYear(YEAR);
     }
 
+    private static String getRandomWord(String str) {
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/OpenSearchClientTest.java:465]: You can replace this with static method `RandomDataGenerator.randomString()` from (no.unit.nva.testutils)

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package no.sikt.nva.nvi.index;
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java:1]: Also update test `shouldReturnPaginatedSearchResultWithCorrectQueryParamsFilterAndQueryInIdIfGiven()` with title?

Comment (monaullah) [reply index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java:1]: You can rename it to `shouldReturnPaginatedSearchResultWithCorrectQueryParamsInIdIfGiven()`

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java [hunk 3]
```diff
@@ -150,7 +150,7 @@ void shouldReturnPaginatedSearchResultWithoutQueryParamFilterInIdIfNotGiven() th
     }
 
     @Test
-    void shouldReturnPaginatedSearchResultWithCorrectQueryParamsFilterAndQueryInIdIfGiven() throws IOException {
+    void shouldReturnPaginatedSearchResultWithCorrectQueryParamsInIdIfGiven() throws IOException {
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java:153]: I would also assert that id contains expected title, like this:
```
@Test
    void shouldReturnPaginatedSearchResultWithCorrectQueryParamsInIdIfGiven() throws IOException {
        mockOpenSearchClient();
        var randomFilter = randomString();
        var randomInstitutions = List.of(randomSiktSubUnit(), randomSiktSubUnit());
        var randomTitle = randomString();
        handler.handleRequest(requestWithInstitutionsAndFilter(randomInstitutions, randomFilter, randomTitle),
                              output, context);
        var response = GatewayResponse.fromOutputStream(output, PaginatedSearchResult.class);
        var paginatedSearchResult = response.getBodyObject(PaginatedSearchResult.class);

        var actualId = paginatedSearchResult.getId().toString();
        var expectedInstitutionQuery = QUERY_PARAM_AFFILIATIONS
                                       + "=" + randomInstitutions.get(0)
                                       + "," + randomInstitutions.get(1);
        var expectedExcludeQuery = QUERY_PARAM_EXCLUDE_SUB_UNITS + "=true";
        var expectedTitleQuery = QUERY_PARAM_TITLE + "=" + randomTitle;

        assertThat(actualId, containsString(QUERY_PARAM_FILTER + "=" + randomFilter));
        assertThat(actualId, containsString(QUERY_PARAM_TITLE + "=" + randomTitle));
        assertThat(actualId, containsString(expectedInstitutionQuery));
        assertThat(actualId, containsString(expectedExcludeQuery));
        assertThat(actualId, containsString(expectedTitleQuery));
    }
```

Comment (EyobSikt) [reply index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java:153]: Is already done. See the code above it.

**Pull request merged**